### PR TITLE
[product-9] Replace converter classes of Product package with Mapstruct

### PR DESCRIPTION
**Pull request by vitaliibredun**
Screenshot 2023-08-18 at 23 21 39
1. Task's name: 
[cart-12] Replace converter classes of Product package with Mapstruct

 2. Developer's full name: 
Vitalii Bredun

 3. Link to the PR: 
https://github.com/Sunagatov/Online-Store/pull/82

 4. Link to Documentation: 
https://github.com/Sunagatov/Online-Store/wiki/API-Specification-for-Product

 5. Changes: 
- Added the Mapstruct interfaces to the converter package.
- Deleted the custom mapper classes.
- Added tests for the Mapstruct.
                                                                                                                                                                                          
 6. Tests: 
ProductInfoDtoConverterTest is checking the mapping.

 7. Test data: 
In the product package of the test package.

 8. Conclusion: 
Project was built with Apache Maven successfully.

### CODE REVIEW

**Review by Sanya124 — APPROVED** [review]

**Review by UladzislauM — APPROVED** [review]

**Review by vpshvd — CHANGES_REQUESTED** [review]

File: src/main/java/com/zufar/onlinestore/product/converter/ProductInfoDtoConverter.java [hunk 1]
```diff
@@ -4,42 +4,23 @@
...
-                pageProductResponseDto.getSize(),
-                pageProductResponseDto.getTotalElements(),
-                pageProductResponseDto.getTotalPages()
-        );
-    }
+    @Mapping(target = "products", source = "pageProductResponseDto.content")
+    @Mapping(target = "page", source = "pageProductResponseDto.number")
+    @Mapping(target = "size", source = "pageProductResponseDto.size")
+    ProductListWithPaginationInfoDto toProductPaginationDto(final Page<ProductInfoDto> pageProductResponseDto);
 }
```
Comment (vpshvd) [inline src/main/java/com/zufar/onlinestore/product/converter/ProductInfoDtoConverter.java:26]: Add empty line at the end of this file

**Review by vpshvd — CHANGES_REQUESTED** [review]

File: src/test/java/com/zufar/onlinestore/ProductInfoDtoConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
+package com.zufar.onlinestore;
```
[the remark this answers is not in the document]
Comment (vitaliibredun) [reply src/test/java/com/zufar/onlinestore/ProductInfoDtoConverterTest.java]: done

File: src/main/java/com/zufar/onlinestore/product/converter/ProductInfoDtoConverter.java (hunk 1 above)
Comment (vitaliibredun) [reply src/main/java/com/zufar/onlinestore/product/converter/ProductInfoDtoConverter.java]: done

**Review by vpshvd — APPROVED** [review]

**Review by AiratGimaev — APPROVED** [review]
Approved

**Review by AiratGimaev — APPROVED** [review]
Approved

**Review by Sunagatov — APPROVED** [review]

**Pull request merged**